### ISSUE TITLE
macOS High Sierra compatibility

### ISSUE THREAD

**Issue by daviswr** [daviswr/ZenPacks.daviswr.OSX.Server.Caching#4]
The Caching service in High Sierra is no longer controlled by serveradmin, requiring modeler and data sources to be updated.

**Comment by daviswr** [comment]
AssetCacheManagerUtil output with Content Caching disabled

```
$ whereis AssetCacheManagerUtil
/usr/bin/AssetCacheManagerUtil

$ AssetCacheManagerUtil status
2018-06-20 12:44:36.391 AssetCacheManagerUtil[2781:318092] Content caching status: {
    Activated = 0;
    Active = 0;
    CacheDetails =     {
    };
    CacheFree = 188442295296;
    CacheLimit = 0;
    CacheStatus = OK;
    CacheUsed = 0;
    Parents =     (
    );
    Peers =     (
    );
    PersonalCacheFree = 188442295296;
    PersonalCacheLimit = 0;
    PersonalCacheUsed = 0;
    Port = 0;
    RegistrationError = "NOT_ACTIVATED";
    RegistrationResponseCode = 403;
    RegistrationStatus = "-1";
    RestrictedMedia = 0;
    ServerGUID = "BA1F5A83-0DF9-4C17-9802-ECEE9C2C37D8";
    StartupStatus = FAILED;
    TotalBytesAreSince = "2018-06-19 14:03:08 +0000";
    TotalBytesDropped = 0;
    TotalBytesImported = 0;
    TotalBytesReturnedToChildren = 0;
    TotalBytesReturnedToClients = 0;
    TotalBytesReturnedToPeers = 0;
    TotalBytesStoredFromOrigin = 0;
    TotalBytesStoredFromParents = 0;
    TotalBytesStoredFromPeers = 0;
}

$ AssetCacheManagerUtil --json status 2>/dev/null
{"name":"status","result":{"CacheLimit":0,"TotalBytesReturnedToClients":0,"StartupStatus":"FAILED","TotalBytesReturnedToPeers":0,"Activated":false,"TotalBytesReturnedToChildren":0,"CacheFree":188442295296,"RegistrationStatus":-1,"RegistrationError":"NOT_ACTIVATED","TotalBytesAreSince":"2018-06-19 14:03:08 +0000","PersonalCacheUsed":0,"TotalBytesDropped":0,"RegistrationResponseCode":403,"CacheStatus":"OK","TotalBytesStoredFromOrigin":0,"CacheDetails":{},"Port":0,"TotalBytesStoredFromPeers":0,"Peers":[],"RestrictedMedia":false,"Parents":[],"PersonalCacheLimit":0,"TotalBytesStoredFromParents":0,"CacheUsed":0,"ServerGUID":"BA1F5A83-0DF9-4C17-9802-ECEE9C2C37D8","TotalBytesImported":0,"PersonalCacheFree":188442295296,"Active":false}}


$ AssetCacheManagerUtil settings
2018-06-20 12:45:07.067 AssetCacheManagerUtil[2783:318349] Content caching settings: {
    AllowPersonalCaching = 1;
    AllowSharedCaching = 1;
    AllowTetheredCaching = 0;
    CacheLimit = 0;
    DataPath = "/Library/Application Support/Apple/AssetCache/Data";
    ListenRangesOnly = 0;
    LocalSubnetsOnly = 1;
    ParentSelectionPolicy = "round-robin";
    PeerLocalSubnetsOnly = 1;
}

$ AssetCacheManagerUtil --json settings 2>/dev/null
{"name":"settings","result":{"ListenRangesOnly":false,"ParentSelectionPolicy":"round-robin","CacheLimit":0,"PeerLocalSubnetsOnly":true,"AllowPersonalCaching":true,"AllowTetheredCaching":false,"DataPath":"\/Library\/Application Support\/Apple\/AssetCache\/Data","LocalSubnetsOnly":true,"AllowSharedCaching":true}}



$ AssetCacheManagerUtil canActivate
2018-06-20 12:45:27.157 AssetCacheManagerUtil[2784:318478] Content caching can be activated.

AssetCacheManagerUtil --json canActivate 2>/dev/null
{"name":"canActivate","result":true}


$ AssetCacheManagerUtil isActivated
2018-06-20 12:45:45.601 AssetCacheManagerUtil[2786:318608] Content caching is deactivated: (no error)

$ AssetCacheManagerUtil --json isActivated 2>/dev/null
{"name":"isActivated","result":false}


$ sudo file "/Library/Application Support/Apple/AssetCache/Data/AssetInfo.db"
/Library/Application Support/Apple/AssetCache/Data/AssetInfo.db: SQLite 3.x database, last written using SQLite version 3019003

$ cat /Library/Preferences/com.apple.AssetCache.plist
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Activated</key>
	<false/>
	<key>CacheLimit</key>
	<integer>10000000000</integer>
	<key>DataPath</key>
	<string>/Library/Application Support/Apple/AssetCache/Data</string>
	<key>LastConfigURL</key>
	<string>http://suconfig.apple.com/resource/registration/v1/config.plist</string>
	<key>LastPort</key>
	<integer>55525</integer>
	<key>LastRegOrFlush</key>
	<date>2018-06-20T19:21:42Z</date>
	<key>LocalSubnetsOnly</key>
	<true/>
	<key>PeerLocalSubnetsOnly</key>
	<true/>
	<key>Port</key>
	<integer>0</integer>
	<key>Region</key>
	<string>0595E5B3C4292427FCF546B29ACEB8FC</string>
	<key>ReservedVolumeSpace</key>
	<integer>2000000000</integer>
	<key>SavedCacheDetails</key>
	<dict>
		<key>Mac Software</key>
		<integer>6343414</integer>
		<key>iOS Software</key>
		<integer>57419745</integer>
	</dict>
	<key>SavedCacheSize</key>
	<integer>63763159</integer>
	<key>ServerGUID</key>
	<string>BA1F5A83-0DF9-4C17-9802-ECEE9C2C37D8</string>
	<key>Version</key>
	<integer>1</integer>
</dict>
</plist>
```

**Comment by daviswr** [comment]
AssetCacheManagerUtil output with Content Caching enabled

```
$ AssetCacheManagerUtil status
2018-06-20 14:29:43.851 AssetCacheManagerUtil[2940:356298] Content caching status: {
    Activated = 1;
    Active = 1;
    CacheDetails =     {
        "Mac Software" = 6343414;
        "iOS Software" = 57419745;
    };
    CacheFree = 9936236841;
    CacheLimit = 10000000000;
    CacheStatus = OK;
    CacheUsed = 63763159;
    Parents =     (
    );
    Peers =     (
    );
    PersonalCacheFree = 10000000000;
    PersonalCacheLimit = 10000000000;
    PersonalCacheUsed = 0;
    Port = 55525;
    PrivateAddresses =     (
        "172.20.88.217"
    );
    PublicAddress = "97.79.186.3";
    RegistrationStatus = 1;
    RestrictedMedia = 0;
    ServerGUID = "BA1F5A83-0DF9-4C17-9802-ECEE9C2C37D8";
    StartupStatus = OK;
    TotalBytesAreSince = "2018-06-20 19:29:29 +0000";
    TotalBytesDropped = 0;
    TotalBytesImported = 0;
    TotalBytesReturnedToChildren = 0;
    TotalBytesReturnedToClients = 0;
    TotalBytesReturnedToPeers = 0;
    TotalBytesStoredFromOrigin = 0;
    TotalBytesStoredFromParents = 0;
    TotalBytesStoredFromPeers = 0;
}

$ AssetCacheManagerUtil settings
2018-06-20 14:30:15.054 AssetCacheManagerUtil[2942:356500] Content caching settings: {
    AllowPersonalCaching = 1;
    AllowSharedCaching = 1;
    AllowTetheredCaching = 0;
    CacheLimit = 10000000000;
    DataPath = "/Library/Application Support/Apple/AssetCache/Data";
    ListenRangesOnly = 0;
    LocalSubnetsOnly = 1;
    ParentSelectionPolicy = "round-robin";
    PeerLocalSubnetsOnly = 1;
}

$ AssetCacheManagerUtil canActivate
2018-06-20 14:30:47.273 AssetCacheManagerUtil[2944:356686] Content caching can be activated.

$ AssetCacheManagerUtil isActivated
2018-06-20 14:30:48.382 AssetCacheManagerUtil[2945:356696] Content caching is activated.

$ cat /Library/Preferences/com.apple.AssetCache.plist
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Activated</key>
	<true/>
	<key>CacheLimit</key>
	<integer>10000000000</integer>
	<key>DataPath</key>
	<string>/Library/Application Support/Apple/AssetCache/Data</string>
	<key>LastConfigURL</key>
	<string>http://suconfig.apple.com/resource/registration/v1/config.plist</string>
	<key>LastPort</key>
	<integer>55525</integer>
	<key>LastRegOrFlush</key>
	<date>2018-06-20T19:29:30Z</date>
	<key>LocalSubnetsOnly</key>
	<true/>
	<key>PeerLocalSubnetsOnly</key>
	<true/>
	<key>Port</key>
	<integer>0</integer>
	<key>Region</key>
	<string>0595E5B3C4292427FCF546B29ACEB8FC</string>
	<key>ReservedVolumeSpace</key>
	<integer>2000000000</integer>
	<key>SavedCacheDetails</key>
	<dict/>
	<key>SavedCacheSize</key>
	<integer>63763159</integer>
	<key>ServerGUID</key>
	<string>BA1F5A83-0DF9-4C17-9802-ECEE9C2C37D8</string>
	<key>Version</key>
	<integer>1</integer>
</dict>
</plist>
```

**Comment by daviswr** [comment]
Closed by 3045253, 02d2cc1, 86dd456
